addons/sourcemod/scripting/vsh/abilities/ability_rage_ghost.sp

```diff
@@ -209,6 +260,60 @@ methodmap CRageGhost < SaxtonHaleBase
 				}
 			}
 			
+			//Building interaction -- you basically just damage them
+			if (this.flBuildingDrain > 0.0)
+			{
+				int iBuilding = MaxClients+1;
+				
+				while ((iBuilding = FindEntityByClassname(iBuilding, "obj_*")) > MaxClients)
+				{
+					bool bLinked = false;
+					if (GetEntProp(iBuilding, Prop_Send, "m_iTeamNum") != iTeam)
+					{
+						float vecTargetOrigin[3];
+						GetEntPropVector(iBuilding, Prop_Send, "m_vecOrigin", vecTargetOrigin);
+						if (GetVectorDistance(vecOrigin, vecTargetOrigin) <= flRadius)
+						{
+							bLinked = true;
+							if (g_flGhostHealStartTime[iClient][iBuilding] == 0.0)
+							{
+								g_flGhostHealStartTime[iClient][iBuilding] = GetGameTime();
+								
+								char sClassname[32];
+								GetEntityClassname(iBuilding, sClassname, sizeof(sClassname));
+								
+								//Teleporters are tiny, so the beam must be down low
+								if (StrEqual(sClassname, "obj_teleporter"))
+									vecTargetOrigin[2] += 5.0;
+								else
+									vecTargetOrigin[2] += 42.0;
+									
+								float vecTargetAngles[3];
+								GetClientAbsAngles(iClient, vecTargetAngles);
+								g_iGhostParticleBeam[iClient][iBuilding] = TF2_SpawnParticle(sParticle[iTeam], vecTargetOrigin, vecTargetAngles, true, iBuilding, EntRefToEntIndex(g_iGhostParticleCentre[iClient]));
+							}
+								
+							float flTimeGap = GetGameTime() - g_flGhostHealStartTime[iClient][iBuilding];
+								
+							int iExpectedSteal = RoundToCeil(flTimeGap * flHealSteal);
```

Comment:
I believe you want `* this.flBuildingDrain` in this math instead of below? Since `iExpectedSteal` is actually what your expecting to deal damage to building